sw/applications/include/layer.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <stdint.h>
+
+typedef enum { CONV2D, GEMM, POOLING, BATCH_NORM } LAYER_TYPE;
+
+typedef enum { RELU, SOFTMAX } ACTIVATION;
+
+typedef enum { FP64 = 8, FP32 = 4, FP16 = 2, FP8 = 1 } PRECISION;
+
+struct layer;
+typedef struct layer_struct layer;
+
+struct layer_struct {
```

Comment:
This is a general layer struct right? So if we only want to run a `GEMM` layer we need this entire struct in memory with all the other members from conv2d and batchnorm that are not used? Could we split this into separate structs `struct layer_gemm` `struct layer_conv2d` etc?

Reply:
yes, that header file is quite outdated and not really necessary. I will change it